macaroni/motorControl/motorControl.ino

```diff
@@ -81,48 +85,48 @@ void setup()
 
 void loop()
 {   
-    muxState = digitalRead(muxStatePin);
-    estop = digitalRead(estopPin);
-    if (estop) {
-        digitalWrite(escMux, 0);
-        digitalWrite(steerMux, 0);
-    } 
-    //if we haven't received a message from the joule in a while, stop driving
-    if (lastMessageTime + 500 < millis()) {
-      timeout = true;
-    }
-    update();
+  muxState = digitalRead(muxStatePin);
+  estop = digitalRead(estopPin);
+  if (estop) {
+    motor(0);
+    steer(0);
+  } 
+  //if we haven't received a message from the joule in a while, stop driving
+  if (lastMessageTime + 500 < millis()) {
+    timeout = true;
+  }
+  update();
 }
 
 void update()
 {
-    if(getMessage()) {
-        desiredSpeed = limitDesiredSpeed(desiredSpeed);
-        desiredHeading = limitDesiredHeading(desiredHeading); 
-    }
-    updateHeading();
-    updateSpeed();
+  if(getMessage()) {
+      desiredSpeed = limitDesiredSpeed(desiredSpeed);
+      desiredHeading = limitDesiredHeading(desiredHeading); 
+  }
+  updateHeading();
+  updateSpeed();
 }
 
-int limitDesiredSpeed(int desiredSpeed) {
-    return min(maxSpeed, max(desiredSpeed, minSpeed));
+int limitDesiredSpeed(float& desiredSpeed) {
+  return min(maxSpeed, max(desiredSpeed, minSpeed));
```

Comment:
Yeah, pick one mechanism. Either modify the float reference or return the result. This doesn't really do either....